api/scheduler_locks_handler.go

```diff
@@ -0,0 +1,94 @@
+package api
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/sirupsen/logrus"
+	"github.com/topfreegames/extensions/middleware"
+	"github.com/topfreegames/maestro/models"
+)
+
+// SchedulerLocksListHandler handler returns schedulers locks
+type SchedulerLocksListHandler struct {
+	App *App
+}
+
+// NewSchedulerLocksListHandler returns an instance of SchedulerLocksListHandler
+func NewSchedulerLocksListHandler(a *App) *SchedulerLocksListHandler {
+	h := &SchedulerLocksListHandler{App: a}
+	return h
+}
+
+func (h *SchedulerLocksListHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	params := schedulerLockParamsFromContext(r.Context())
+	l := middleware.GetLogger(r.Context())
+	logger := l.WithFields(logrus.Fields{
+		"source":    "schedulerLocksList",
+		"operation": "get locks",
+		"scheduler": params.SchedulerName,
+	})
+	logger.Debug("Getting scheduler locks")
+	locksPrefixes := []string{h.App.Config.GetString("watcher.lockKey")}
+	locks, err := models.ListSchedulerLocks(
+		h.App.RedisClient.Trace(r.Context()), params.SchedulerName, locksPrefixes,
```

Comment:
Can't you pass the context to the function instead of wrapping the redis client?

Reply:
why would that be better?

Reply:
I just think it's ugly for the handler to know of redis, but passing the context wouldn't fix this anyway, given that the handler is already the one managing the client :(

So yeah, I don't think it would matter much

Reply:
Agreed on everything